File: ABC/131/D.py
n = int(input())
a = []

for _ in range(n):
    x, y = map(int, input().split())
    a.append((x, y))

a.sort(key=lambda x:x[1])
curr = 0

for i in range(n):
    x, y = a[i]
    curr += x
    if curr > y:
        print("No")
        exit()

print("Yes")

File: ABC/131/D.cpp
#include <iostream>
#include <vector>

using namespace std;

#define rep(i, n) for (int i = 0; i < n; i++)
using P = pair<int, int>;
using ll = long long;

int main() {
  int n;
  cin >> n;
  vector<P> p(n);
  rep(i, n) cin >> p[i].second >> p[i].first;
  sort(p.begin(), p.end());
  ll curr = 0;

  rep(i, n) {
    curr += p[i].second;
    if (curr > p[i].first) {
      cout << "No" << endl;
      return 0;
    }
  }
  cout << "Yes" << endl;
}
